src/FINDOLOGIC/Helpers/ParameterBuilder.php

```diff
@@ -0,0 +1,449 @@
+        }
+
+        $this->addParam(self::PUSH_ATTRIB, [$key => [$value => $factor]], self::ADD_VALUE);
+        return $this;
+    }
+
+    /**
+     * Sets the count param. It is used to set the number of products that should be displayed.
+     *
+     * @see https://docs.findologic.com/doku.php?id=integration_documentation:request#limiting_paging_parameters
+     * @param $value int
+     * @return ParameterBuilder
+     */
+    public function setCount($value)
+    {
+        if (!is_integer($value)) {
+            throw new InvalidParamException(self::COUNT);
+        }
+
+        $this->addParam(self::COUNT, $value);
+        return $this;
+    }
+
+    /**
+     * Sets the first param. It is used for pagination between pages. If your count is for example 20 and you switch
+     * to the second page, set this parameter to 20 --> the first product on the next page. Do not set the parameter to
+     * 21, because the product listing is 0-based.
+     *
+     * @param $value int
+     * @return ParameterBuilder
+     */
+    public function setFirst($value)
+    {
+        if (!is_integer($value)) {
+            throw new InvalidParamException(self::FIRST);
+        }
+
+        $this->addParam(self::FIRST, $value);
+        return $this;
+    }
+
+    /**
+     * Sets the identifier param. It is used to display only the item that is given. The query param is ignored.
+     *
+     * @param $value string ID of the item.
+     * @return ParameterBuilder
+     */
+    public function setIdentifier($value)
+    {
+        if (!is_string($value)) {
+            throw new InvalidParamException(self::IDENTIFIER);
+        }
+
+        $this->addParam(self::IDENTIFIER, $value);
+        return $this;
+    }
+
+    /**
+     * Adds the group param. It is used to show only products for one or more specific groups.
+     *
+     * @param $value string
+     * @return ParameterBuilder
+     */
+    public function addGroup($value)
+    {
+        if (!is_string($value)) {
+            throw new InvalidParamException(self::GROUP);
+        }
+
+        $this->addParam(self::GROUP, ['' => $value], self::ADD_VALUE);
+        return $this;
+    }
+
+    /**
+     * Adds the forceOriginalQuery param. It is used for Smart Did You Mean.
+     *
+     * @return ParameterBuilder
+     */
+    public function setForceOriginalQuery()
+    {
+        $this->addParam(self::FORCE_ORIGINAL_QUERY, 1);
+        return $this;
+    }
+
+    /**
+     * Adds an own param to the parameter list. This can be useful if you want to put some special key value pairs to
+     * get a different response from FINDOLOGIC.
+     * IMPORTANT: Both $key or $value are NOT validated and NOT tested. **Using this is neither recommended nor
+     * supported.**
+     *
+     * @param $key mixed
+     * @param $value mixed
+     * @param $method string Use ParameterBuilder::ADD_VALUE to add the param and not overwrite existing ones and
```

Comment:
Well.. https://github.com/phpDocumentor/phpDocumentor2/issues/557 exists since `5 Aug 2012` and there was no progress since this time. Some people like "[anurbol](https://github.com/anurbol)" also say

> These type of issues are the reason why I gave up with PHP after 11 years and moved to node.js + Typescript. PHP is moving too slow for 2018

sad that this is not implemented this yet after 6 years since the issue was submitted.

-1👎 for PHPDoc.